### PR TITLE
add issues collection support

### PR DESCRIPTION
**Pull request by RobiNino**

### CODE REVIEW

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/ArtifactoryBuildInfoClient.java [hunk 1]
```diff
@@ -81,6 +79,8 @@
     public static final String APPLICATION_VND_ORG_JFROG_ARTIFACTORY_JSON = "application/vnd.org.jfrog.artifactory+json";
     public static final String APPLICATION_JSON = "application/json";
     public static final String ITEM_LAST_MODIFIED = "/api/storage/";
+    private static final String LATEST = "LATEST";
```
Comment (yahavi) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/ArtifactoryBuildInfoClient.java]: Looks like LATEST and LAST_RELEASE are not in use

File: build-info-api/src/main/java/org/jfrog/build/api/IssueTracker.java [hunk 2]
```diff
@@ -37,4 +38,22 @@ public String getVersion() {
...
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof IssueTracker)) {
+            return false;
+        }
+
+        IssueTracker that = (IssueTracker) o;
+
+        if (!Objects.equals(name, that.name)) {
```
Comment (yahavi) [inline build-info-api/src/main/java/org/jfrog/build/api/IssueTracker.java]: Can be simplified to: 
return (Objects.equals(name, that.name) && Objects.equals(version, that.version));

File: build-info-api/src/main/java/org/jfrog/build/api/Vcs.java [hunk 3]
```diff
@@ -41,6 +42,31 @@ public String toString() {
...
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Vcs)) {
+            return false;
+        }
+
+        Vcs that = (Vcs) o;
+
+        if (!Objects.equals(revision, that.revision)) {
```
Comment (yahavi) [inline build-info-api/src/main/java/org/jfrog/build/api/Vcs.java]: Can be simplified to:
return (Objects.equals(revision, that.revision) && Objects.equals(url, that.url));

File: build-info-api/src/main/java/org/jfrog/build/api/IssuesCollectionConfig.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
+package org.jfrog.build.api;
+
+import org.apache.commons.lang.StringUtils;
+
+import java.io.IOException;
+import java.io.Serializable;
+
+public class IssuesCollectionConfig implements Serializable {
```
Comment (eyalbe4) [inline build-info-api/src/main/java/org/jfrog/build/api/IssuesCollectionConfig.java]: I suggest adding a description for class, describing its role.

File: build-info-api/src/main/java/org/jfrog/build/api/IssuesCollectionConfig.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        }
+        if (issues.keyGroupIndex == 0) {
+            throw new IOException(ISSUES_COLLECTION_ERROR_PREFIX + MISSING_CONFIGURATION_ERROR + "keyGroupIndex");
+        }
+        if (issues.summaryGroupIndex == 0) {
+            throw new IOException(ISSUES_COLLECTION_ERROR_PREFIX + MISSING_CONFIGURATION_ERROR + "summaryGroupIndex");
+        }
+    }
+
+    public static class Issues implements Serializable {
```
Comment (eyalbe4) [inline build-info-api/src/main/java/org/jfrog/build/api/IssuesCollectionConfig.java]: I suggest adding a description for each of this class fields.

File: build-info-api/src/main/java/org/jfrog/build/api/Vcs.java [hunk 6]
```diff
@@ -41,6 +42,26 @@ public String toString() {
                 ", url='" + url + '\'';
     }
 
+    @Override
+    public boolean equals(Object o) {
```
Comment (eyalbe4) [inline build-info-api/src/main/java/org/jfrog/build/api/Vcs.java]: Is this method still needed, following the changes in the Jenkins Plugin code base?

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/ArtifactoryBuildInfoClient.java [hunk 7]
```diff
@@ -190,6 +188,11 @@ public void sendBuildInfo(String buildInfoJson) throws IOException {
     }
 
     public Build getBuildInfo(String buildName, String buildNumber) throws IOException {
+        buildNumber = getLatestBuildNumberFromArtifactory(buildName, buildNumber);
```
Comment (eyalbe4) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/ArtifactoryBuildInfoClient.java]: I suggest adding the following comment above this line:
// Only If the value of the buildNumber sent is "LATEST" or "LAST_RELEASE", replace the value with a specific build number.

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/issuesCollection/IssuesCollector.java [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+        Set<Issue> affectedIssues = new HashSet<>();
+
+        Pattern pattern = Pattern.compile(issuesConfig.getIssues().getRegexp());
+        String[] lines = gitLog.split("\\R");
+        for (String line : lines) {
+            Matcher matcher = pattern.matcher(line);
+            if (matcher.find()) {
+                Issue foundIssue = getMatchingIssue(keyIndex, summaryIndex, matcher, issuesConfig);
+                affectedIssues.add(foundIssue);
+                logger.debug("Found issue: " + foundIssue.getKey());
```
Comment (eyalbe4) [inline build-info-extractor/src/main/java/org/jfrog/build/extractor/issuesCollection/IssuesCollector.java]: I suggest changing this from *debug* to *info* and change the message to:
"Added issue XXX to the build-info."

**Pull request merged**